Source/SwiftLintFramework/Rules/FirebaseCoreRule.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  FirebaseCoreRule.swift
+//  SwiftLint
+//
+//  Created by Ibrahim Ulukaya (Google Inc.) on 3/8/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+
+import SourceKittenFramework
+
+public struct FirebaseCoreRule: ConfigurationProviderRule, OptInRule {
+
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "firebase_core",
+        name: "Firebase Core",
+        description: "Firebase should be configured before use.",
+        nonTriggeringExamples: [
+            "class AppDelegate: UIResponder, UIApplicationDelegate {\n" +
+            "  func application(_ application: UIApplication," +
+            "      didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {\n" +
+            "    FirebaseApp.configure()\n" +
+            "    return true \n" +
+            "  }\n" +
+            "}"
+        ],
+        triggeringExamples: [
+            "class AppDelegate: UIResponder, UIApplicationDelegate {\n" +
+            "  func application(_ application: UIApplication," +
+            "      didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {\n" +
+            "    return true \n" +
+            "  }\n" +
+            "}"
```

Comment:
I now check if there is helper function doing it as well. So checking all the way down in AST. Finding if there is a function getting created there, that would do this in it's initialization would be complicated.

I'll leave this rule for the suggested use case, hence it's an opt-in rule, which only returns warning, that should be enough coverage. The rule is towards new devs that could forget to initialize these objects before use. The dev who's already initializing with a special class would already be aware of it, hence won't need the rule or just ignore the warning.